commands/priority_test.go

```diff
@@ -49,3 +49,60 @@ func Test_getTemplateStoreURL(t *testing.T) {
 		})
 	}
 }
+
+func Test_getOverrideNamespace(t *testing.T) {
+	const defaultFunctionNamespace = defaultFunctionNamespace
+	failCases := 0
+
+	tests := []struct {
+		stack    string
+		flag     string
+		want     string
+		scenario string
+	}{
+		// Test cases
+		{
+			stack:    "",
+			flag:     "",
+			want:     defaultFunctionNamespace,
+			scenario: "no namespace value set in flag or in namespace field of stack file",
+		},
+
+		{
+			stack:    "openfaas-fn",
+			flag:     "default",
+			want:     "default",
+			scenario: "stack file provides namespace as 'openfaas-fn' while CLI flag provides namespace as 'default'",
+		},
+
+		{
+			stack:    "default",
+			flag:     "",
+			want:     "default",
+			scenario: "stack file provides namespace as 'default' whereas no namespace is provided by CLI",
+		},
+
+		{
+			stack:    "",
+			flag:     "default",
+			want:     "default",
+			scenario: "flag provides namespace as 'default' whereas no namespace is provided by stack file",
+		},
+	}
+
+	// Run the test for each test case defined in "tests"
+	for _, testCase := range tests {
+		testCase := testCase
+		functionNamespace := getNamespace(testCase.flag, testCase.stack)
+
+		t.Run(testCase.scenario, func(t *testing.T) {
+			if functionNamespace != testCase.want {
+				t.Errorf("Expected Namespace: %s Got: %s\n", testCase.want, functionNamespace)
```

Comment:
It's fine if you want to fail immediately here, you can use `t.Fatalf`, no need to try to make it complicated.

The error also doesn't take into the account the feedback I gave you, it should be `namespace incorrect want: %s, but got %s`, even better would be `%q` to show any whitespace.